cli/tests/pcluster/api/controllers/test_cluster_operations_controller.py

```diff
@@ -958,11 +964,84 @@ def test_incompatible_version(self, client, mocker):
+                            "Please check /var/log/chef-client.log in the head node, or "
+                            "check the chef-client.log in CloudWatch logs. Please refer to "
+                            "https://docs.aws.amazon.com/parallelcluster/latest/ug/"
+                            "troubleshooting-v3.html#troubleshooting-v3-get-logs for more "
+                            "details on ParallelCluster logs.' for "
+                            "uniqueId: i-01fdb3c2b73182693",
+                        },
+                    ]
+                ],
+                [{"failureCode": "FsxMountFailure", "failureReason": "Failed to mount FSX."}],
             ),
-            ("CREATE_FAILED", [], None),
+            (
+                "CREATE_FAILED",
+                [
+                    [
+                        {
+                            "StackId": "fake-id",
+                            "StackName": "fake-name",
+                            "ResourceType": "AWS::CloudFormation::WaitCondition",
+                            "ResourceStatus": "CREATE_FAILED",
+                            "ResourceStatusReason": "WaitCondition timed out. Received 0 conditions when "
+                            "expecting 1",
+                        },
+                    ]
+                ],
+                [{"failureCode": "HeadNodeBootstrapFailure", "failureReason": "Cluster creation timed out."}],
+            ),
+            (
+                "CREATE_FAILED",
+                [
+                    [
+                        {
+                            "StackId": "fake-id",
+                            "StackName": "fake-name",
+                            "ResourceType": "AWS::CloudFormation::WaitCondition",
+                            "ResourceStatus": "CREATE_FAILED",
+                            "ResourceStatusReason": "Resource creation cancelled",
+                        },
+                    ]
+                ],
+                [
+                    {
+                        "failureCode": "ResourceCreationFailure",
+                        "failureReason": "Failed to create resources for head node bootstrap.",
+                    }
+                ],
+            ),
+            (
+                "CREATE_FAILED",
+                [],
+                [{"failureCode": "ClusterCreationFailure", "failureReason": "Failed to create the cluster."}],
+            ),
+        ],
+        ids=[
+            "get_stack_events_without_next_token",
+            "get_stack_events_with_next_token",
+            "fail_to_mount_fsx",
+            "cluster_creation_timeout",
+            "cluster_resource_creation_failure",
+            "no_stack_event",
         ],
-        ids=["get_stack_events_without_next_token", "get_stack_events_with_next_token", "no_stack_event"],
     )
     def test_cluster_creation_failed(
         self, client, mocker, cfn_stack_status, get_stack_events_response, expected_failure_reason
```

Comment:
nitpick, I'd change also `expected_failure_reason ` to `expected_failures` to be more clear

Reply:
Good idea! changed